File: ApplicationLLA.Server/Models/Message.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ApplicationLLA.Server.Models
{
    public class Message
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int MessageId { get; set; }

        public required int ConversationId { get; set; }

        [StringLength(128)]
        public required string SenderId { get; set; }

        [StringLength(128)]
        public required string ReceiverId { get; set; }

        public bool IsDeletedFromSender { get; set; }
        public bool IsDeletedFromReceiver { get; set; }

        [StringLength(300)]
        public required string Content { get; set; }
        public required DateTime SentDate { get; set; }

        public required Conversation Conversation { get; set; }
    }

}

File: frontend/src/Models/Message.ts
export type ConversationGet = {
    conversationId: number;
    userId: string;
    userFullName: string;
    lastReadDate: Date;
    numberOfUnreadMessages: number;
    pictureLink: string;
}
export type MessagePost = {
    conversationId: number;
    content: string;
}

export type Message = {
    messageId: number;
    senderId: string;
    content: string;
    sentDate: Date;
}
